From: osmo-remsim build script <build@osmo-remsim>
Subject: Add --disable-examples configure option

Add a configure option to optionally disable building the examples
and utils directories. This is useful for:
1. Cross-compilation scenarios where examples/utils may have linking issues
2. Embedded/OpenWRT builds where examples/utils are not needed
3. Faster builds when only the library is required

The examples and utils directories are still built by default to maintain
backward compatibility.

--- a/Makefile.am
+++ b/Makefile.am
@@ -1,7 +1,15 @@
 AUTOMAKE_OPTIONS = foreign dist-bzip2 1.6
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = include src examples utils tests
+if ENABLE_EXAMPLES
+EXAMPLES_DIR = examples
+endif
+
+if ENABLE_EXAMPLES
+UTILS_DIR = utils
+endif
+
+SUBDIRS = include src $(EXAMPLES_DIR) $(UTILS_DIR) tests
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libosmo-netif.pc
--- a/configure.ac
+++ b/configure.ac
@@ -114,6 +114,13 @@ AS_IF([test "x$ENABLE_LIBSCTP" = "xyes"], [
 	LIBS=$old_LIBS
 ])
 
+AC_ARG_ENABLE([examples],
+	[AS_HELP_STRING([--disable-examples],
+		[Disable building example programs (default: enabled)])],
+	[enable_examples=$enableval],
+	[enable_examples=yes])
+AM_CONDITIONAL(ENABLE_EXAMPLES, test "x$enable_examples" = "xyes")
+
 found_pcap=yes
 AC_CHECK_HEADERS(pcap.h,,found_pcap=no)
 AM_CONDITIONAL(HAVE_PCAP, test "$found_pcap" = yes)
